app/internal/pay/client.go

```diff
@@ -21,6 +23,32 @@ type Client struct {
 	HttpClient Doer
 }
 
+type GovUKPayTime time.Time
+
+func (g *GovUKPayTime) UnmarshalJSON(b []byte) error {
+	s := strings.Trim(string(b), "\"")
+	s = strings.Trim(s, "0Z")
+
+	t, err := time.Parse(time.RFC3339, s+"00Z")
+	if err != nil {
+		return err
+	}
+
+	*g = GovUKPayTime(t)
+	return nil
```

Comment:
This is really gross - I couldn't work out how to get Go to parse a date string in the format`2016-01-21T17:15:000Z`. `RFC3339` was closest but still slightly out.

Reply:
In the Pay docs they say it is like `2022-03-25T13:11:29.019Z`, which would be `time.RFC3339Nano`, so I think you can simplify this to

```go

func (g *GovUKPayTime) UnmarshalText(b []byte) error {
	t, err := time.Parse(time.RFC3339Nano, string(b))
	if err != nil {
		return err
	}

	*g = GovUKPayTime(t)
	return nil
}

func (g GovUKPayTime) MarshalText() ([]byte, error) {
	return []byte(g.Format(time.RFC3339Nano)), nil
}

func (g GovUKPayTime) Format(s string) string {
	return time.Time(g).Format(s)
}
```

(also using Un/MarshalText is easier to deal with since you are inside the string, I think the only time you'd not want to use it is when you have to deal with weird null behaviour, :eyes: Sirius...)

Reply:
It would also be nice if we the structs exposed from this package used normal `time.Time` and we translated in the package, but happy for you to leave if it seems like too much hassle right now